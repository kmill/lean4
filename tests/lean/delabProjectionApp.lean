/-!
# Delaboration of projection functions
-/

structure A where
  x : Nat

structure B extends A where
  y : Nat

structure C extends B where
  z : Nat

variable (a : A) (b : B) (c : C)

section
/-!
Checking projection delaboration, including parent projection collapse.
-/

#check a.x
#check b.x
#check c.x

#check b.y
#check c.y

#check c.z

end

section
/-!
Checking `pp.structureProjections` can turn off this delaborator.
-/

set_option pp.structureProjections false

#check a.x
#check b.x
#check c.x

#check b.y
#check c.y

#check c.z

end
