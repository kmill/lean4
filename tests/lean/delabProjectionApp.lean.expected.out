a.x : Nat
b.x : Nat
c.x : Nat
b.y : Nat
c.y : Nat
c.z : Nat
A.x a : Nat
A.x (B.toA b) : Nat
A.x (B.toA (C.toB c)) : Nat
B.y b : Nat
B.y (C.toB c) : Nat
C.z c : Nat
